src/main/java/clzzz/helper/commons/util/DateTimeUtil.java

```diff
@@ -21,4 +22,71 @@
     public static final String BACK_UP_PATTERN = "yyyyMMdd_HH_mm_ss";
     public static final DateTimeFormatter BACK_UP_FORMAT =
             DateTimeFormatter.ofPattern(BACK_UP_PATTERN);
+
+    /**
+     * Check if the given date is a valid date.
+     */
+    public static boolean isValidDateString(String date) {
+        String[] parsedDate = date.split("/");
+        int day = Integer.parseInt(parsedDate[0]);
+        int month = Integer.parseInt(parsedDate[1]);
+        int year = Integer.parseInt(parsedDate[2]);
+        return isValidDate(day, month, year);
+    }
+
+    /**
+     * Check if the given time is a valid time.
+     */
+    public static boolean isValidTimeString(String time) {
+        int timeInt = Integer.parseInt(time);
+        if (timeInt < 0 || timeInt >= 2400) {
+            return false;
+        }
+        if (timeInt % 100 >= 60) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Check if the date is valid
+     */
+    public static boolean isValidDate(int day, int month, int year) {
+        if (day < 1 || day > 31 || month < 1 || month > 12) {
+            return false;
+        }
+        if (day <= 28) {
+            return true;
+        }
+        switch (month) {
+        case 2:
+            if (isLeap(year)) {
+                return day <= 29;
+            }
+            return false;
+        case 4:
+        case 6:
+        case 9:
+        case 11:
+            return day <= 30;
+        default:
+            return true;
+        }
```

Comment:
LocalDate#lengthOfMonth()

Reply:
thx! I go change it

Reply:
I can't use isLeapYear() or lengthOfMonth() cuz that needs me to convert the string to Localdate first. But if I want to keep the padding, I need to stick to LINIENT/SMART resolverstyle, and that will possibly change the year/month in the adapted localdate. That will affect the result of the functions u mentioned. I think it is better to just manually check it.

Reply:
If we follow ab3, we’d always have a wrapper class and check with a regex.
But now for some attributes we either just use library classes or don’t use regex for validation.
I find it strange that on the one hand we choose to use the library classes for convenience, on the other hand we insist on rewriting a parser for them.

Reply:
just use STRICT? I believe dd accepts both 1 and 01

Reply:
no I checked alr, it doesn't...